### DISCUSSION TITLE
Options, Ci and ISO hosting

### DISCUSSION THREAD

**Discussion by FacundoAcevedo** [SpiralLinux/SpiralLinux-project#511] [General]
Hi there!

I've been happily testing SpiralLinux in a VM, and I'll install it as my main distro soon.  
I have some ideas and questions I'd like to share:

1. **Custom Installation Options**  
   Would it be possible to make Calamares ask if I want to opt out of certain options/packages during installation?  
   For example:
   - Not installing VirtualBox-related packages.
   - Enabling Wayland.
   - Skipping proprietary drivers.

2. **CI for Building the Distro**  
   Have you considered using a CI/CD pipeline for building SpiralLinux?  
   I think it would make the process more transparent, allowing people to easily fork and contribute to this great project.  

3. **Alternative ISO Hosting**  
   Regarding ISO hosting, have you thought about also using [The Internet Archive](https://archive.org)?  
   They provide direct downloads and automatic WebTorrent support, which could be a helpful addition.

Thank you for the time and effort you've put into this project!

**Comment by geckolinux** [comment]
Hi there, glad to hear that you like SpiralLinux, and thanks for the suggestions.

> Would it be possible to make Calamares ask if I want to opt out of certain options/packages during installation?

Technically it would be a matter of making Calamares uninstall certain packages. The SpiralLinux ISO is basically just a full installation of SpiralLinux on a compressed read-only filesystem, which Calamares then decompresses onto the target disk. It doesn't use DEB packages as a means of installation. However after dumping it onto the target disk Calamares does `chroot` into the installed system and run a few automatic `apt remove` operations for packages that are not relevant to the installed system. It looks like https://github.com/calamares/calamares/tree/calamares/src/modules/packagechooser has a UI for presenting package choices to the user, but I'm not sure if it would allow for batch uninstallation of packages based on the user's selection. And specifically regarding these options:
- > Not installing VirtualBox-related packages.
    - This would be a useful improvement
- > Enabling Wayland
    - Nope. https://github.com/orgs/SpiralLinux/discussions/183#discussioncomment-4184401
- > Skipping proprietary drivers
    - SpiralLinux is focused on pragmatism and "just working". The average user doesn't care if their drivers are proprietary or not, they just want their hardware to work. I'm not saying that the issues around proprietary drivers aren't important, but it's not a cause that I'm interested in fighting for and ultimately SpiralLinux can't do anything to change the situation in the grand scheme of things. Users that know enough to care about these issues can easily remove the proprietary drivers on their personal installations.

> Have you considered using a CI/CD pipeline for building SpiralLinux?

I honestly don't understand how that all works. I doubt that the pipeline process would be practical if it even works at all, because SpiralLinux requires a specifically configured Debian VM to run `live-build` with internet access (for installing the base Flatpak support and some Debian packages too that pull resources from the internet with `curl`). I personally hate Git and all of the hassle that goes along with it. SpiralLinux is specifically intended to be a low-maintenance project that I tinker with on my spare time and only when an updated set of ISOs is needed. I just use a local development directory on my computer with the build recipe, and then `rsync` that directory to a VPS to build and upload the ISOs to Sourceforge. From there it depends 100% on Debian's infrastructure for everything that the installed system needs so I don't have to work on it for a while.

> Regarding ISO hosting, have you thought about also using [The Internet Archive](https://archive.org/)?

Given their recent downtime I have serious doubts about their sustainability. Also in my experience direct download speeds from archive.org are abysmally slow, and I had always assumed that it was mainly intended just for distributing unmaintained software that is no longer available directly from the original developer. I don't think they have download mirrors either do they? But please correct me if I'm wrong.

Thanks again for the complements and good suggestions.

**Reply by FacundoAcevedo** [reply]
Thanks for the detailed answers

With the bitorrent download I got  5 mirrors seeding thru webseed 
![image](https://github.com/user-attachments/assets/5938c673-3775-442c-82ec-a1f1fb6621ee)

I tried some times with the direct download but I ended up always getting the file from the same subdomain.

Edit: Just noticed that the screenshot has only four seed, but just before taking it I got five :P

**Reply by geckolinux** [reply]
Thanks for the reply. So that's for torrents, and what about direct downloads?

**Reply by FacundoAcevedo** [reply]
well... I got 1.5 Mbps
